src/java.base/share/classes/jdk/internal/foreign/MemorySessionImpl.java

```diff
@@ -255,15 +254,16 @@ public final void run() {
         }
 
         static void cleanup(ResourceCleanup first) {
-            CleanupException pendingException = null;
+            RuntimeException pendingException = null;
             ResourceCleanup current = first;
             while (current != null) {
                 try {
                     current.cleanup();
-                } catch (Throwable ex) {
+                } catch (RuntimeException ex) {
                     if (pendingException == null) {
-                        pendingException = new CleanupException(ex);
-                    } else {
+                        pendingException = ex;
+                    } else if (ex != pendingException) {
+                        // note: self-suppression is not supported
```

Comment:
Does the current test cover the `ex == pendingException` case?

Reply:
No, it does not